### ISSUE TITLE
Form submission cannot be resolved - regression introduced between 1.0.0-next.519 (working) and 1.0.0-next.522 (broken)

### ISSUE THREAD

**Issue by kyllerss** [sveltejs/kit#7387]
### Describe the bug

**TLDR;** I have tracked down a breaking change when I updated my SvelteKit dependency from `1.0.0-next.519` (working) to `1.0.0-next.522` (broken). I have `+page.server.ts` that exports both a `load` handler and a default `actions` handler. When I POST to the action, I get a 404 page with the message `Function not found...`. The logs complain of `Missing form submission function handler` as it iterates through different variants of the endpoint. Issue is still happening with the latest SvelteKit version (`1.0.0-next.525`).

**Here are the specifics:**
I have an action (`src/routes/(dashboard)/home/pets/upload/+page.server.ts`):
```js

import { type RequestEvent, redirect } from "@sveltejs/kit";
import type { Actions } from './$types';

/** @type {import('./restricted/upload').PageServerLoad} */
export async function load(event: RequestEvent) {

    // <some work here>

    return {a: true, b: 1};
}

export const actions: Actions = {
    default: async (event: RequestEvent) => {

        // <handle post data>

        throw redirect(302, "/home/pets/upload/finish");
    }
};
```

The form that performs the post looks like this (`src/routes/(dashboard)/home/pets/upload/+page.svelte`): 

```html
        <form method="POST">
            <input ... />
            <input type="submit" value="Submit"/>
        </form>
```

When I attempt to perform a POST to it, I get a 404. 

### Reproduction

Working on creating one (though someone stop me if they know where things went wrong). 

### Logs

The console logs show the following:

```shell
docker-app-1     |  ›   Warning: Missing form submission function handler
docker-app-1     | Request from ::ffff:172.18.0.5: POST /home/pets/upload
docker-app-1     | Response with status 404 in 2 ms.
docker-app-1     |  ›   Warning: Missing form submission function handler
docker-app-1     | Request from ::ffff:172.18.0.5: POST /home/pets/upload.html
docker-app-1     | Response with status 404 in 0 ms.
docker-app-1     |  ›   Warning: Missing form submission function handler
docker-app-1     | Request from ::ffff:172.18.0.5: POST /home/pets/upload.htm
docker-app-1     | Response with status 404 in 0 ms.
docker-app-1     |  ›   Warning: Missing form submission function handler
docker-app-1     | Request from ::ffff:172.18.0.5: POST /home/pets/upload/index.html
docker-app-1     | Response with status 404 in 0 ms.
docker-app-1     |  ›   Warning: Missing form submission function handler
docker-app-1     | Request from ::ffff:172.18.0.5: POST /home/pets/upload/index.htm
docker-app-1     | Response with status 404 in 0 ms.
```

The network headers are (after some minor cleanup):
```
POST /home/pets/upload HTTP/1.1
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.5
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 70
Content-Type: application/x-www-form-urlencoded
Cookie: __cflb=XXXX; csrf_token_xxxx=xxxxxx=; xxxxxxxxxxxxx
Host: localhost:4173
Origin: http://localhost:4173
Pragma: no-cache
Referer: http://localhost:4173/home/pets/upload
Sec-GPC: 1
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36

HTTP/1.1 404 Not Found
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: http://localhost:4173
Access-Control-Expose-Headers: Set-Cookie, Cache-Control, Expires, Last-Modified, Pragma, Content-Length, Content-Language, Content-Type
Date: Mon, 24 Oct 2022 22:16:19 GMT
Strict-Transport-Security: max-age=0;
Vary: Origin
X-Powered-By: Express
Content-Length: 21
Content-Type: text/plain; charset=utf-8
```

### System Info

```Shell
System:
    OS: Linux 5.15 KDE neon 5.26 5.26
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
    Memory: 7.49 GB / 15.34 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 19.0.0 - ~/apps/node-current/bin/node
    npm: 8.19.2 - ~/apps/node-current/bin/npm
  Browsers:
    Brave Browser: 106.1.44.112
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.84 
    @sveltejs/adapter-netlify: ^1.0.0-next.67 => 1.0.0-next.81 
    @sveltejs/kit: next => 1.0.0-next.525 
    svelte: ^3.44.0 => 3.52.0 
    vite: ^3.1.0 => 3.1.8
```

### Severity

blocking an upgrade

**Comment by dummdidumm** [comment]
The `Missing form submission function handler` warning is not coming from SvelteKit. Is see that's coming from a log inside Docker. Is the form post working in dev mode (without Docker in the mix)?

**Comment by kyllerss** [comment]
Sorry, I should have clarified that bit. Yes, I am hosting the SvelteKit node instance in a docker container. I've been having a hard time reproducing the issue on a fresh/paired down install. I'm currently stepping through the SvelteKit code trying to find where things break down. I'll report more once I find the culprit.

**Comment by kyllerss** [comment]
Narrowing it down a little further. Seems to be something relating to Netlify. I enabled debug logging on node and saw that the POST request was being routed directly to the port being used for background functions. 

I'm going to ping someone on their side for ideas.

**Comment by kyllerss** [comment]
I was finally able to create a minimal reproduction of the issue. Seems to be a Netlify issue. I am have opened [an issue](https://github.com/netlify/cli/issues/4873) on their side. Closing this bug entry and tracking there. I'll re-open this issue if it turns out to somehow be a SvelteKit issue.
